types/aws-lambda/test/cloudwatch-tests.ts

```diff
@@ -43,3 +43,7 @@ const scheduledHandler: ScheduledHandler = async (event, context, callback) => {
     callback();
     callback(new Error());
 };
+
+const scheduledHandlerWithDetails: ScheduledHandler<{ status: string }> = async (event, context, callback) => {
+    const eventDetail: ScheduledEvent<{ status: string }> = event.detail;
```

Comment:
I'm confused: shouldn't this be just `{ status: string }`?

Reply:
I think I misunderstood the scheduled event type. You are right. I'll give an update tomorrow.

Reply:
I've fix the type on ScheduleHandler, however this will make it a breaking change.

Reply:
I could be missing something, but I don't see what the breaking change would be?

Reply:
Any existing consumer of the interface ScheduledHandler it may have an impact if they were using the detail type.

Before this one was valid:

```ts
const scheduleHandler: ScheduledHandler = async(event) => {
        const e: ScheduleEvent = event.details // it looks wrong, it should be type of detail no the type of ScheduleEvent
     const id: string = event.details.id// id should be in the event.
     const details: any = event.details.details // a bit odd no?
}
```

After this changes

```ts
const scheduleHandler: ScheduledHandler = async(event) => {
        const e: any = event.details
        const id: string = event.id
}
```

Reply:
... that's not a breaking change, as the existing obviously wrong, but compiling code will still compile fine, because `TDetail` will default to `any`.

Reply:
Ah, no, the existing code was simply wrong, and it's just that nobody has complained yet. This fixes it.